jazz_metrics/index.js

```diff
@@ -0,0 +1,303 @@
+            } else if(error.result === "notFoundError"){
+                return cb(JSON.stringify(errorHandler.throwNotFoundError(error.message)));
+            } else if(error.result === "unauthorized") {
+                return cb(JSON.stringify(errorHandler.throwUnauthorizedError(error.message)));
+            } else{
+                return cb(JSON.stringify(errorHandler.throwInternalServerError("Error in fetching cloudwatch metrics" )));
+            }
+        });
+    } catch (e) {
+        return cb(JSON.stringify(errorHandler.throwInternalServerError("Error in fetching cloudwatch metrics")));
+    }
+
+};
+
+function genericValidation(event) {
+    return new Promise((resolve, reject) => {
+        if (!event && !event.body) {
+            reject({
+                result:"inputError",
+                message: "Invalid Input Error"
+            });
+        }
+        if(!event.principalId) {
+            reject({
+                result: "unauthorized",
+                message: "Unauthorized"
+            });
+        }
+
+        resolve();
+    });
+}
+
+function getAssetsDetails(config, eventBody) {
+    return new Promise((resolve, reject) => {
+        var asset_api_payload = {
+            "service" : eventBody.service,
+            "domain": eventBody.domain,
+            "environment" : eventBody.environment
+        };
+        var asset_api_options = {
+            url: config.SERVICE_API_URL + config.ASSETS_URL,
+            headers: {
+                "Content-Type": "application/json",
+            },
+            method: "POST",
+            rejectUnauthorized: false,
+            requestCert: true,
+            async : true,
+            json: true,
+            body: asset_api_payload,
+        };
+
+        logger.info("asset_api_options :- " + JSON.stringify(asset_api_options));
+
+
+        request(asset_api_options, function (error, response, body) {
+            if(error) {
+                reject(error);
+            } else {
+                var apiAssetsArray = body.data;
+                if(apiAssetsArray && apiAssetsArray.length === 0){
+                    logger.error("Assets not found for this service, domain, environment. ", JSON.stringify(asset_api_options) );
+                    resolve([]);
+                } else {
+                    var userStatistics = eventBody.statistics.toLowerCase();
+
+                    // Massaging data from assets api , to get required list of assets which contains type, asset_name and statistics.
+                    var assetsArray = utils.getAssetsObj(apiAssetsArray,userStatistics);
+                    resolve(assetsArray);
+                }
+            }
+        });
+    });
+}
+
+function validateAssets(assetsArray, eventBody) {
+    return new Promise((resolve, reject) => {
+        if(assetsArray.length > 0){
+            var newAssetArray = [];
+            var invalidTypeCount = 0;
+            assetsArray.forEach(function(assetItem){
```

Comment:
-indentation.
-would be nice to have new line here

Reply:
- too much going in inside the loop here. please simplify

Reply:
Done